e2e/nomostest/gitproviders/repository.go

```diff
@@ -332,46 +361,72 @@ func (g *Repository) Get(path string) client.Object {
 		// runtime.Object but not a metav1.Object.
 		// Most registered objects should be metav1.Object.
 		// But if they aren't, we need to be able to read their metadata.
-		return uObj
+		return uObj, nil
 	}
 
 	// Convert the unstructured object to a typed object
 	err = runtime.DefaultUnstructuredConverter.FromUnstructured(uObj.UnstructuredContent(), obj)
 	if err != nil {
-		g.T.Fatal(err)
+		return nil, errors.Wrap(err, "converting typed object to unstructured")
+	}
+	return obj, nil
+}
+
+// MustGet calls Get and fails the test on error, logging the result.
+func (g *Repository) MustGet(t testing.NTB, path string) client.Object {
```

Comment:
If we have this `MustGet` abstraction here, it seems like we could either keep the default behavior of the other methods or define a `Must` alternative (e.g. `MustCommitAndPush`). WDYT?

Is the reason for this method that `nt.Must` swallows the return value?